src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PosBlockSignatureRule.cs

```diff
@@ -51,19 +51,24 @@ private bool CheckBlockSignature(PosBlock block)
                 return res;
             }
 
-            if (block.BlockSignature.IsEmpty())
+            return CheckPOSSignature(block.BlockSignature, block.GetHash(), block.Transactions[1], this.Logger);
+        }
+
+        public static bool CheckPOSSignature(BlockSignature signature, uint256 blockHash, Transaction coinStake, ILogger logger)
+        {
+            if (signature.IsEmpty())
             {
-                this.Logger.LogTrace("(-)[EMPTY]:false");
+                logger.LogTrace("(-)[EMPTY]:false");
                 return false;
             }
 
-            TxOut txout = block.Transactions[1].Outputs[1];
+            TxOut txout = coinStake.Outputs[1];
 
             if (PayToPubkeyTemplate.Instance.CheckScriptPubKey(txout.ScriptPubKey))
             {
                 PubKey pubKey = PayToPubkeyTemplate.Instance.ExtractScriptPubKeyParameters(txout.ScriptPubKey);
-                bool res = pubKey.Verify(block.GetHash(), new ECDSASignature(block.BlockSignature.Signature));
-                this.Logger.LogTrace("(-)[P2PK]:{0}", res);
+                bool res = pubKey.Verify(blockHash, new ECDSASignature(signature.Signature));
+                logger.LogTrace("(-)[P2PK]:{0}", res);
```

Comment:
Why change the logger style?

Reply:
I suspect his passing in the logger from the caller so that we can know that this, for example, came from `ProvenHeaderCoinstakeRule`.

Reply:
Is it not that `CheckPOSSignature` has been made static to allow calling it from elsewhere?

Reply:
@zeptin, correct.

Reply:
@fassadlr , yes this makes it clear which rule is involved.

Reply:
This is not correct, if you want to pass a logger move the method in to stake validator.

Reply:
Actually StakeValidator has its own logger just use that one

Reply:
Done.

Reply:
Fixed